backend/manager/modules/bll/src/main/java/org/ovirt/engine/core/bll/snapshots/RemoveSnapshotSingleDiskLiveCommand.java

```diff
@@ -128,9 +128,18 @@ public boolean performNextOperation(int completedChildCount) {
                 // If the images were already merged, just add the orphaned image
                 getParameters().setMergeStatusReturnValue(synthesizeMergeStatusReturnValue());
             }
-            nextCommand = buildDestroyCommand(ActionType.DestroyImage, getActionType(),
-                    new ArrayList<>(getParameters().getMergeStatusReturnValue().getImagesToRemove()));
             getParameters().setNextCommandStep(RemoveSnapshotSingleDiskStep.REDUCE_IMAGE);
+            if (getParameters().isImageErrorOccurredOnMerge()) {
+                log.info("Merge verified, although image not found during merge. Skipping destroy image command " +
+                                "execution. Storage Type: '{}', Disk: '{}' Snapshot: '{}'",
+                        getStorageDomain().getStorageType(),
+                        getImage().getName(),
+                        getImage().getDescription());
+                setCommandStatus(CommandStatus.SUCCEEDED);
```

Comment:
do I get this correctly that in case the image doesn't exist we would still call MergeExtend and not reduce the image afterwards? if so, would it be possible to check, with get vol info or something similar, whether the image exists in advance and if it's not to skip everything?

Reply:
Would it be sufficient? I was exactly thinking about this option in the context of ColdMerge, but here we do check the volume chain. So the question is what might happen besides the "positive" flow here - might there be a snapshot in the DB, but otherwise the system is in the state where if we run `syncDbRecords` we will make things even worse?
The other option is to proceed to REDUCE after skipping DESTROY.

Reply:
@bennyz WDYT?

Reply:
I think we can skip directly to fixing the DB, and then maybe check the chain with MergeStatusCommand?

Reply:
The DB is fixed in RemoveSnapshotSingleDiskLiveCommand.onSucceed() for the image and then in RemoveSnapshotCommand.endVmCommand() for the snapshot.

Reply:
so if syncDbRecords fails in onSucceeded we could mark the child command as failed, and in turn RemoveSnapshotCommand will fail too, no?

Reply:
So, what we want to do is:
1. Before MergeExtend we run VolumeInfo to check if disk exists
2. If not, we skip Extend, Merge straight to MergeStatus, and if it succeeds we finish the command successfully
3. Fail if syncDbRecords fail (although it shouldn't)
?

Reply:
yes, unless a reason it won't work it should both save unnecessary operations in vdsm, and have a fail safe in case something's wrong